tests/test_scheduler.py

```diff
@@ -0,0 +1,139 @@
+        net.param_init_fn = self.param_init_fn
+        super().__init__(module=net, num_classes=num_classes)
+
+        # Important: It is crucial that the FC layers are bound to `self`
+        # for the optimizer surgery tests.
+        # These tests attempt to perform surgery on `fc1` layer, and we want
+        # to make sure that post-surgery, self.fc1 refers to the same parameters
+        # as self.net[1]
+        self.fc1 = fc1
+        self.fc2 = fc2
+
+    def param_init_fn(self, module: torch.nn.Module):
+        init_fn = partial(torch.nn.init.normal_, mean=0.0, std=0.1)
+
+        if isinstance(module, torch.nn.Linear):
+            init_fn(module.weight)
+            if module.bias is not None:
+                torch.nn.init.zeros_(module.bias)
+
+
+@pytest.fixture
+def dummy_schedulers_state(request: pytest.FixtureRequest):
+    device = None
+    for item in request.session.items:
+        device = DeviceCPU(
+        ) if item.get_closest_marker('gpu') is None else DeviceGPU()
+        break
+    assert device != None
+    state = State(
+        model=SimpleModel(),
+        run_name='run_name',
+        device=device,
+        rank_zero_seed=17,
+        max_duration=MAX_DURATION,
+    )
+    state.set_dataloader([None] * STEPS_PER_EPOCH, 'train')
+    return state
+
+
+@pytest.mark.parametrize('scheduler,ssr,test_times,expected_lrs', [
+    pytest.param(
+        InverseSquareRootWithWarmupScheduler(t_warmup='10ba',
+                                             t_scale='10ba',
+                                             t_cooldown='0ba',
+                                             alpha_f_decay=0,
+                                             alpha_f_cooldown=0), 1.0,
+        ['0ba', '5ba', '10ba', '40ba', '90ba', '100ba'],
+        [0.0, 0.5, 1.0, 0.5, 0.33333, 0.31623]),
+    pytest.param(
+        InverseSquareRootWithWarmupScheduler(t_warmup='20ba',
+                                             t_scale='2ba',
+                                             t_cooldown='10ba',
+                                             alpha_f_decay=0.4,
+                                             alpha_f_cooldown=0.1), 1.0,
+        ['0ba', '10ba', '20ba', '36ba', '90ba', '95ba', '100ba'],
+        [0.0, 0.5, 1.0, 0.6, 0.5, 0.3, 0.1]),
+])
+def test_scheduler_init(scheduler: ComposerScheduler, ssr: float,
+                        test_times: List[str], expected_lrs: List[float],
+                        dummy_schedulers_state: State):
+
+    state = dummy_schedulers_state
+    assert state.dataloader_len is not None
+    assert state.max_duration is not None
+    state.max_duration = Time(value=int(state.max_duration.value * ssr),
+                              unit=state.max_duration.unit)
+    for test_time, expected_lr in zip(test_times, expected_lrs):
+        parsed_time = Time.from_timestring(test_time)
+        assert parsed_time.unit in [TimeUnit.EPOCH, TimeUnit.BATCH]
+        if parsed_time.unit == TimeUnit.EPOCH:
```

Comment:
did you intend to test epochs as well? If not, can just delete this code branch.

Reply:
Did not intend to test epochs so I deleted this code branch. This function was lifted directly from composer scheduler tests so I don't want to change it too much so that it is straightforward to move the tests over when we upstream to composer.